crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1105,6 +1105,54 @@ fn find_only_token_in_range(
     token
 }
 
+/// Attach an enclosed end-of-line comment to a [`StmtImportFrom`].
+///
+/// For example, given:
+/// ```python
+/// from foo import (  # comment
+///    bar,
+/// )
+/// ```
+///
+/// The comment will be attached to the `StmtImportFrom` node as a dangling comment, to ensure
+/// that it remains on the same line as the `StmtImportFrom` itself.
+fn handle_import_from_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    // The comment needs to be on the same line, but before the first member. For example, we want
+    // to treat this as a dangling comment:
+    // ```python
+    // from foo import (  # comment
+    //     bar,
+    //     baz,
+    //     qux,
+    // )
+    // ```
+    // However, this should _not_ be treated as a dangling comment:
+    // ```python
+    // from foo import (bar,  # comment
+    //     baz,
+    //     qux,
+    // )
+    // ```
+    // Thus, we check if the first non-trivia token _before_ the comment is an `LParen`.
+    if comment.line_position().is_end_of_line() {
+        let mut tokens =
+            SimpleTokenizer::up_to_without_back_comment(comment.start(), locator.contents())
+                .skip_trivia();
```

Comment:
Nit: I would love to remove the backward lexing if possible in the future. It kind of works, but I'm worried that we'll run into issues sooner or later (e.g. magic commands are interesting...) . Could we instead search forward from the end of the identifier (or the start of the statement if absent)?

Reply:
I think I figured out a solution that doesn't require lexing. If the comment is end-of-line, and _between_ the start of the statement and the first member, it _has_ to be immediately following the parenthesis.